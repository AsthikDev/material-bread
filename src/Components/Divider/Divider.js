import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import withTheme from '../../Theme/withTheme';

class Divider extends Component {
  static propTypes = {
    style: PropTypes.object,
    theme: PropTypes.object,
    marginVertical: PropTypes.number,
  };
  render() {
    const { style, theme, marginVertical, ...rest } = this.props;

    return (
      <View
        style={[
          theme.divider,
          { marginVertical: marginVertical ? marginVertical : 8 },
          style,
        ]}
        {...rest}
      />
    );
  }
}

export default withTheme(Divider);
