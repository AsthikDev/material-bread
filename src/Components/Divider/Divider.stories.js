import React from 'react';

import { Divider } from '../../';
import Header from '../../storybook/components/Header';
import Container from '../../storybook/components/Container';
import { storiesOf } from '../../storybook/helpers/storiesOf';

storiesOf('Components|Divider', module)
  .addParameters({ jest: ['Divider'] })
  .add('Simple', () => (
    <Container>
      <Header title={'Divider'} />

      <Divider />
      <Divider />
      <Divider marginVertical={24} />
    </Container>
  ));
