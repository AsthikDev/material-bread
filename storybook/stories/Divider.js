import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import { BreadProvider, Divider } from '../../src/index';
import Header from '../components/Header';

storiesOf('Divider', module)
  .addDecorator(story => <View style={{ padding: 16 }}>{story()}</View>)
  .add('Simple', () => (
    <BreadProvider value={{}}>
      <Header>Divider</Header>
      <Divider />
      <Divider />
      <Divider marginVertical={24} />
    </BreadProvider>
  ));
