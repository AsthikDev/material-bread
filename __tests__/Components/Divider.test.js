import React from 'react';
import Divider from '../../src/Components/Divider';

import renderer from 'react-test-renderer';

test('it renders Divider', () => {
  const tree = renderer.create(<Divider />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders Divider with marginVertical', () => {
  const tree = renderer.create(<Divider marginVertical={16} />).toJSON();
  expect(tree).toMatchSnapshot();
});
