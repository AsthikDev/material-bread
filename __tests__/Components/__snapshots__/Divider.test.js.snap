// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Divider 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "height": 0.5,
      "marginVertical": 8,
      "opacity": 0.12,
      "width": "100%",
    }
  }
/>
`;

exports[`it renders Divider with marginVertical 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "height": 0.5,
      "marginVertical": 16,
      "opacity": 0.12,
      "width": "100%",
    }
  }
/>
`;
